client/app/templates/components/show-project/actions-list/ulurp-number-link.hbs

```diff
@@ -0,0 +1,29 @@
+{{#if @action.dcpUlurpnumber}}
```

Comment:
Why is this its own component and not other parts of markup in actions-list? What if you rolled this back into actions-list, and looked at it more broadly: you loop over actions, there's some delegation behavior with `if`s, and use build-url consistently.

What you're proposing here looks like an ad hoc decision — it "feels" like it's "a lot" of markup, with some state (through relativeUrl), which I've pointed out seems unnecessary, so therefore, it should be its own component. Why? Why not make dcpZoningresolution and dcpCcresolutionnumber into their own components? I know they're tiny, but at least it'd make sense the second time going into it.

Reply:
Even more confusing is the fact that the `if` block is evaluated in the component for the situation itself, rather than where every other type is evaluated, inside the actions-list. I would look at this again and try to imagine how someone who has never seen the code before would think this through.

Reply:
>  and looked at it more broadly: you loop over actions, there's some delegation behavior with ifs, and use build-url consistently.
I don't follow what you mean here by "delegation behavior with ifs..." 

> it "feels" like it's "a lot" of markup, with some state (through relativeUrl), which I've pointed out seems unnecessary, so therefore, it should be its own component. Why? 
Good observation. I think that was part of it. I was following our discussions from building long forms in Applicant Portal -- we observed a ton of unwieldy, lengthy templates, and thought it would be better if we could break these up into smaller components so they're more manageable and testable. 
However I can also see your point that this is "componentizing just because".

Reply:
@allthesignals 
> Even more confusing is the fact that the if block is evaluated in the component for the situation itself, 

 Sorry I don't follow this 🤔

Reply:
The template looks like this:

```
if (x) do this
if (y) do this
runSomethingThatEvaluates(z)
if (a) do this
```
rather than

```
if (x) do this
if (y) do this
if (z) do runSomething()
if (a) do this
```

It doesn't follow the pattern suggested in its context.

Reply:
> I don't follow what you mean here by "delegation behavior with ifs..."

Sorry, I just mean that it renders something based on a conditional. It delegates the responsibility of rendering markup based on the conditional.

Reply:
Oh I see, so with both commentsyou're saying  it reads better when every "if" is visible at the parent template. It'll be clearer to readers that the pattern applies for each of those displayed properties of projectAction. 

Thanks!